lib/fakeBlockChain.js

```diff
@@ -2,9 +2,16 @@ const Buffer = require('safe-buffer').Buffer
 const utils = require('ethereumjs-util')
 
 module.exports = {
-  getBlock: function (n, cb) {
-    // FIXME: this will fail on block numbers >53 bits
-    var hash = utils.sha3(Buffer.from(utils.bufferToInt(n).toString(), 'utf8'))
+  getBlock: function (blockTag, cb) {
+    var hash
+
+    if (Buffer.isBuffer(blockTag)) {
+      hash = utils.sha3(blockTag)
+    } else if (Number.isInteger(blockTag)) {
+      hash = utils.sha3(utils.toBuffer(blockTag).toString('hex'))
```

Comment:
This need a pre-pending `0x`, otherwise resulting hashes are different:

```
> var blockTag=3
undefined
> Number.isInteger(blockTag)
true
> utils.sha3(utils.toBuffer(blockTag).toString('hex'))
<Buffer a6 33 ed b4 4b 8c eb e0 45 ec f3 b7 3d 0a 30 7a 03 52 39 b2 36 c7 7f 3a 27 ce eb db 1d f2 e1 e6>
> blockTag=Buffer.from([blockTag])
<Buffer 03>
> utils.sha3(blockTag)
<Buffer 69 c3 22 e3 24 8a 5d fc 29 d7 3c 5b 05 53 b0 18 5a 35 cd 5b b6 38 67 47 51 7e f7 e5 3b 15 e2 87>
> blockTag=3
3
> utils.sha3('0x' + utils.toBuffer(blockTag).toString('hex'))
<Buffer 69 c3 22 e3 24 8a 5d fc 29 d7 3c 5b 05 53 b0 18 5a 35 cd 5b b6 38 67 47 51 7e f7 e5 3b 15 e2 87>
```

Reply:
I'd argue it makes no difference though as the code path is not used at the moment. All what matters is the hash is consistent across calls.

Reply:
If it makes no difference I will just add that, I find that inconsistent and not so beautiful and feel more comfortable when it's adjusted.

Reply:
It keeps the old inconsistent behaviour :)